Libraries/Blob/Blob.js

```diff
@@ -0,0 +1,156 @@
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule Blob
+ * @flow
+ */
+'use strict';
+
+const { BlobModule } = require('NativeModules');
+
+import type { BlobProps } from './BlobTypes';
+
+const UUID4 = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx';
+function createUUID(): string {
+  /* eslint-disable no-bitwise */
+  return UUID4.replace(/[xy]/g, (pat) => {
+    let nibble = Math.random() * 16 | 0;
+    if (pat === 'y') {
+      nibble &= 0x3;
+      nibble |= 0x8;
+    }
+    return nibble.toString(16);
+  });
+}
+
+/**
+ * Opaque JS representation of some binary data in native.
+ *
+ * The API is modeled after the W3C Blob API, with one caveat
+ * regarding explicit deallocation. Refer to the `close()`
+ * method for further details.
+ *
+ * Example usage in a React component:
+ *
+ *   class WebSocketImage extends React.Component {
+ *      state = {blob: null};
+ *      componentDidMount() {
+ *        let ws = this.ws = new WebSocket(...);
+ *        ws.binaryType = 'blob';
+ *        ws.onmessage = (event) => {
+ *          if (this.state.blob) {
+ *            this.state.blob.close();
+ *          }
+ *          this.setState({blob: event.data});
+ *        };
+ *      }
+ *      componentUnmount() {
+ *        if (this.state.blob) {
+ *          this.state.blob.close();
+ *        }
+ *        this.ws.close();
+ *      }
+ *      render() {
+ *        if (!this.state.blob) {
+ *          return <View />;
+ *        }
+ *        return <Image source={{uri: URL.createObjectURL(this.state.blob)}} />;
+ *      }
+ *   }
+ */
+class Blob {
+  /**
+   * Size of the data contained in the Blob object, in bytes.
+   */
+  size: number;
+  /*
+   * String indicating the MIME type of the data contained in the Blob.
+   * If the type is unknown, this string is empty.
+   */
+  type: string;
+
+  /*
+   * Unique id to identify the blob on native side (non-standard)
+   */
+  blobId: string;
+  /*
+   * Offset to indicate part of blob, used when sliced (non-standard)
+   */
+  offset: number;
+
+  /**
+   * Construct blob instance from blob data from native.
+   * Used internally by modules like XHR, WebSocket, etc.
+   */
+  static create(props: BlobProps): Blob {
+    return Object.assign(Object.create(Blob.prototype), props);
+  }
+
+  /**
+   * Constructor for JS consumers.
+   * Currently we only support creating Blobs from other Blobs.
+   */
+  constructor(parts: Array<Blob>, options: any) {
+    let blobId = createUUID();
+    let size = 0;
+    parts.forEach((part) => {
+      if (!(part instanceof Blob)) {
+        throw new Error('Can currently only create a Blob from other Blobs');
+      }
+      size += part.size;
+    });
+    BlobModule.createFromParts(parts, blobId);
+    return Blob.create({
+      blobId,
+      offset: 0,
+      size,
+    });
+  }
+
+  slice(start?: number, end?: number): Blob {
```

Comment:
Yes, it's the index of last byte. This just follows the Blob API. Will add a link https://developer.mozilla.org/en-US/docs/Web/API/Blob/slice